src/app/child-dev-project/children/child-details/grouped-child-attendance/grouped-child-attendance.component.ts

```diff
@@ -39,9 +44,31 @@ export class GroupedChildAttendanceComponent implements OnInit {
 
   private async loadActivities() {
     this.loading = true;
-    this.activities = (
-      await this.attendanceService.getActivitiesForChild(this.entity.getId())
-    ).filter((a) => !a.excludedParticipants.includes(this.entity.getId()));
+    const allActivities = await this.attendanceService.getActivitiesForChild(
+      this.entity.getId(),
+    );
+
+    this.activities = allActivities.filter(
+      (a) =>
+        !a.excludedParticipants.includes(this.entity.getId()) &&
+        a.isActive == true,
+    );
+
+    this.archivedActivities = allActivities.filter(
+      (a) =>
+        !a.excludedParticipants.includes(this.entity.getId()) &&
+        a.isActive == false,
+    );
+
     this.loading = false;
   }
+
+  onActivityChange(selectedArchivedActivity: RecurringActivity) {
+    this.selectedActivity = true;
+    this.selectedArchivedActivities = this.archivedActivities.filter(
+      (a) =>
+        a.title == selectedArchivedActivity.title &&
+        !a.excludedParticipants.includes(this.entity.getId()),
+    );
+  }
```

Comment:
see my other comment, please don't rely on the title as there may be duplicates.

I suggest
```suggestion
    this.selectedActivity = selectedArchivedActivity;
  }
```

(with the relevant type and template changes, of course)

Reply:
updated the template and ts file